### ISSUE TITLE
Add option to output to Markdown table

### ISSUE THREAD

**Issue by peterjaap** [pyrech/composer-changelogs#69]
We use Gitlab instead of Github, and Gitlab uses markdown for issues. It would be great to have an option to have the output of this plugin generated as a Markdown table, so we can copy/paste it into an issue.

**Comment by pyrech** [comment]
Hello, thanks for your suggestion.

I like the idea, I don't think implementing it would be too complicated. We could either change the default output or just add an option to allow the choice of the desired output.

Would you like to give it a try?

**Comment by peterjaap** [comment]
Changing the default output to a Markdown table wouldn't look pretty in the console. So I'm in favor of adding an option, although I'm not sure how to pass that option to the plugin without having to pass a flag everytime.

**Comment by pyrech** [comment]
Indeed, let's have another optional table output instead :+1: Speaking of it, which columns do you need? It would be better to have the same columns for the three operations (install, update, uninstall).

About the architecture, I think we could add support for a new option in the plugin section of the composer.json. See [this documentation](https://github.com/pyrech/composer-changelogs/blob/master/doc/configuration.md) to have a look at the existing config supported by this plugin with the [`ConfigBuilder`](https://github.com/pyrech/composer-changelogs/blob/master/src/Config/ConfigBuilder.php) and the related [`Config` DTO](https://github.com/pyrech/composer-changelogs/blob/master/src/Config/Config.php).

I guess we could add an `OutputFormatter` interface with 2 implementations (current formatting and table). Then, according to the new config option, we would pass the correct implementation to the `OutputFormatter` / `getOutput` method of `OperationHandler` classes.

Does this sounds good to you?

**Comment by peterjaap** [comment]
Yes, sounds good!

The table columns would preferably be package name, previous version, current version, release notes link, compare link. So something like this; https://github.com/davidrjonas/composer-lock-diff#example-plain-text-table-output

**Comment by peterjaap** [comment]
@pyrech come to think of it, your suggestion would mean I'd have to edit the composer.json file in *every* project I'd like to see a table output, which in our case is a lot (we run an agency). Isn't there a way to store parameters like this somewhere in composer's global scope, like in a file somewhere?

**Comment by pyrech** [comment]
Yes, that is supported :slightly_smiling_face:. The plugin configuration can also be set in your global location (usually in `~/.composer/composer.json`) :wink:. See https://github.com/pyrech/composer-changelogs/blob/master/doc/configuration.md#location

**Comment by peterjaap** [comment]
@pyrech of course, forgot about that one. Perfect!
